src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -71,22 +78,39 @@ internal DocumentInfo(DocumentAttributes attributes, TextLoader loader)
             string filePath = null,
             bool isGenerated = false)
         {
-            return new DocumentInfo(new DocumentAttributes(id, name, folders, sourceCodeKind, filePath, isGenerated), loader);
+            return Create(id, name, folders, sourceCodeKind, loader, filePath, isGenerated, documentServiceProvider: null);
+        }
+
+        internal static DocumentInfo Create(
```

Comment:
❓ Why not make this overload the one with optional parameters, and remove the optional parameter values from the method above?

Reply:
existing Create is public one. so we are not going to change yet. and we will make everything as internal first until we have preview bits out and we settle on everything. when we RTM, we will see whether we want to make these public API so external (razor, xaml and etc) can provide this without IVT.